delivery_auto_refresh/models/sale_order.py

```diff
@@ -13,14 +13,18 @@ class SaleOrder(models.Model):
         compute="_compute_carrier_id", store=True, readonly=False
     )
 
-    @api.depends("partner_id")
+    @api.depends("partner_id", "partner_shipping_id")
     def _compute_carrier_id(self):
         if hasattr(super(), "_compute_carrier_id"):
             super()._compute_carrier_id()
-        for rec in self:
-            if rec.partner_id:
-                if rec.partner_id.property_delivery_carrier_id:
-                    rec.carrier_id = rec.partner_id.property_delivery_carrier_id
+        # Compute the carrier as Odoo does in its core
+        # https://github.com/OCA/OCB/blob/27edc28a2c19fad80053e668a8c72e7a629e4c1a
+        # /addons/delivery/models/sale_order.py#L68-L71
+        for rec in self.filtered("partner_shipping_id"):
+            rec.carrier_id = (
+                rec.partner_shipping_id.property_delivery_carrier_id
+                or rec.partner_shipping_id.commercial_partner_id.property_delivery_carrier_id
+            ).filtered("active")
```

Comment:
What do you think about calling the same method and extract the carrier and being compatible with possible extensions?

```suggestion
        for order in self:
            action = order.action_open_delivery_wizard()
            order.carrier_id = action["context"]["default_carrier_id"]
```

Reply:
I like it. Let's go for it then